frontend/components/puzzle3.jsx

```diff
@@ -1,34 +1,80 @@
 import React from 'react';
-
 import PuzzleHeader from './puzzleHeader';
+import { AnswerAwareDiv } from './puzzle.jsx';
 
-// Answer: Your name
+// Answer: [5,4,3,2,1]
 
+const initial_combo = ['','','','',''];
 class Puzzle3 extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleClear = this.handleClear.bind(this);
+    this.submitAnswer = this.submitAnswer.bind(this);
+    this.state = {
+      combo : initial_combo
+    };
+  }
+
+
+  handleClick(input) {
+    return () => {
+      var temparray = this.state.combo;
+      var flag = 'full';
+      for (var i = 0; i <= temparray.length; i++) {
+        if ( temparray[i] === '') {
+          this.setState(state => {
+            const combo = state.combo.slice();
+            combo.splice(i, 1, input);
+            return { combo };
+          });
+          flag = 'good';
+          break;
+        }
+      }
+      if (flag === 'full') {
+        console.log('too many numbers');
+      }
+    };
+  }
+  submitAnswer() {
+    var userinput = this.state.combo;
+    this.props.send('submit', { puzzle: '3', answer: userinput });
+  }
+
+  handleClear() {
+    this.setState({ combo:initial_combo });
+    this.props.reset();
+  }
+
   render() {
+    const pw = this.state.combo;
     return (
       <div>
         <PuzzleHeader title="Puzzle Three" />
-        <form id="puzzle3-form">
-          <p>
-            What belongs to you but other people use it more than you?
-          </p>
-          <p>
-            <input type="text" name="puzzleAnswer" autoComplete="off" placeholder="Type here" />
-          </p>
-
-          <div>
-            <button type="submit">submit</button>
-          </div>
-          <div>
-            <button type="submit">submit</button>
-          </div>
-          <div>  
-            <button type="submit">submit</button>
-
-          </div>
-        </form>
-
+        <div className="riddle">
+          <p>Some riddle pertaining to the pictures below</p>
+          <p>make the correct selections are you&apos;ll best this foe</p>
+        </div>
+        <div>
+          {pw.map(pw => <AnswerAwareDiv correct={this.props.correct}> {pw} </AnswerAwareDiv>)}
+        </div>
+        <div className="image-keypad-container">
```

Comment:
i think we should move the styles from css for this class into a component, but we can do that in a different PR